app/src/backlight.c

```diff
@@ -142,23 +143,42 @@ int zmk_backlight_set_brt(int brt) {
     return zmk_backlight_save_state();
 }
 
-int zmk_backlight_get_brt() { return state.on ? state.brightness : 0; }
+int zmk_backlight_toggle() { return state.on ? zmk_backlight_off() : zmk_backlight_on(); }
 
-int zmk_backlight_toggle() { return zmk_backlight_set_on(!state.on); }
-
-int zmk_backlight_on() { return zmk_backlight_set_on(true); }
+int zmk_backlight_set_brt(uint8_t brightness) {
+    if (brightness > BRT_MAX) {
+        return -ENOTSUP;
```

Comment:
I think this function shouldn't return an error, consider the following situation:
- `state.brightness = 100`
- `state.on = false`
- press `BL_INC`

Expected behavior: turn on the backlight.
Actual behavior: an error is returned.

Reply:
Changed it to set the on state properly. The set brightness function is meant primarily for syncing between both sides of a split

Reply:
The new change also doesn't work, if `state.on = false`, you can set the brightness to any value greater than `BRT_MAX`.

Reply:
Because the `set_brt` function works in conjunction with the `calc_brt` function which automatically auto ranges to within the max min values when using the `BL_INC` the only reason the `set_brt` function would receive an out of range brightness  is if the user configures their keymap to do so, I'm not sure the best way to deal with this is just clamp the brightness regardless but I can change it